drivers/serial/usart_mcux_lpc.c

```diff
@@ -83,6 +88,158 @@ static int usart_mcux_lpc_err_check(struct device *dev)
 	return err;
 }
 
+#ifdef CONFIG_UART_INTERRUPT_DRIVEN
+static int usart_mcux_lpc_fifo_fill(struct device *dev, const u8_t *tx_data,
+			       int len)
+{
+	const struct usart_mcux_lpc_config *config = dev->config->config_info;
+	u8_t num_tx = 0U;
+
+	while ((len - num_tx > 0) &&
+	       (USART_GetStatusFlags(config->base)
+		& kUSART_TxFifoEmptyFlag)) {
```

Comment:
Wouldn't `kUSART_TxFifoNotFullFlag` be better here?

Reply:
@MaureenHelm you are right

Reply:
You updated fifo_read to use `kUSART_RxFifoNotEmptyFlag`, but you didn't update fifo_fill to use `kUSART_TxFifoNotFullFlag` yet.